integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/timeseries/TestTimeSeriesMatchStrategySuite.scala

```diff
@@ -0,0 +1,294 @@
+  }
+
+  val timeSeries = "preaggregate"
+
+  test("test timeseries match 1: select small size when there are >1 timeseries match") {
+
+    dropDataMaps("maintable", "agg1_minute", "agg2_minute")
+    sql(
+      s"""
+         | create datamap agg1 on table mainTable
+         | using '$timeSeries'
+         | DMPROPERTIES (
+         |   'timeseries.eventTime'='mytime',
+         |   'timeseries.hierarchy'='minute=1')
+         | as select mytime, sum(age), count(age), max(age), min(age)
+         | from mainTable
+         | group by mytime
+      """.stripMargin)
+
+    val df1 = sql(
+      """
+        | select
+        |   timeseries(mytime,'minute') as minuteLevel,
+        |   sum(age) as sum
+        | from mainTable
+        | where
+        |     timeseries(mytime,'minute')<'2016-02-23 01:02:00' and
+        |     timeseries(mytime,'minute')>='2016-02-23 01:01:00'
+        | group by
+        |   timeseries(mytime,'minute')
+        | order by
+        |   timeseries(mytime,'minute')
+      """.stripMargin)
+
+    checkAnswer(df1, Seq(Row(Timestamp.valueOf("2016-02-23 01:01:00"), 60)))
+
+    preAggTableValidator(df1.queryExecution.analyzed, "maintable_agg1_minute")
+
+    sql(
+      s"""
+         | create datamap agg2 on table mainTable
+         | using '$timeSeries'
+         | DMPROPERTIES (
+         |   'timeseries.eventTime'='mytime',
+         |   'timeseries.hierarchy'='minute=1')
+         | as select mytime, sum(age)
+         | from mainTable
+         | group by mytime
+      """.stripMargin)
+
+    val df2 = sql(
+      """
+        | select
+        |   timeseries(mytime,'minute') as minuteLevel,
+        |   sum(age) as sum
+        | from mainTable
+        | where
+        |     timeseries(mytime,'minute')<'2016-02-23 01:02:00' and
+        |     timeseries(mytime,'minute')>='2016-02-23 01:01:00'
+        | group by
+        |   timeseries(mytime,'minute')
+        | order by
+        |   timeseries(mytime,'minute')
+      """.stripMargin)
+
+    checkAnswer(df2, Seq(Row(Timestamp.valueOf("2016-02-23 01:01:00"), 60)))
+
+    preAggTableValidator(df2.queryExecution.analyzed, "maintable_agg2_minute")
+  }
+
+  test("test timeseries match 2: select small size when there are >1 timeseries match") {
+    dropDataMaps("maintable", "agg1_minute", "agg2_minute")
+    sql(
+      s"""
+         | create datamap agg1 on table mainTable
+         | using '$timeSeries'
+         | DMPROPERTIES (
+         |   'timeseries.eventTime'='mytime',
+         |   'timeseries.hierarchy'='minute=1')
+         | as select mytime, sum(age), count(age), max(age), min(age)
+         | from mainTable
+         | group by mytime
+      """.stripMargin)
+
+    sql(
+      s"""
+         | create datamap agg2 on table mainTable
+         | using '$timeSeries'
+         | DMPROPERTIES (
+         |   'timeseries.eventTime'='mytime',
+         |   'timeseries.hierarchy'='minute=1')
+         | as select mytime, sum(age)
```

Comment:
should create this datamap first, and modify testcase title

Reply:
ok, done